### ISSUE TITLE
minor graphical issue when extract assets.

### ISSUE THREAD

**Issue by iseeaitch** [spelunky-fyi/modlunky2#304]
When extracting assets with modlunky, there seems to be a problem with the sprites being slightly deformed.
Translucent parts(especially glowing effect) on a extracted image turns black.
This can be clearly seen at Arrow of light, jellyfish, orb, etc.

![0](https://user-images.githubusercontent.com/106856709/171982656-f2384a61-2b30-4da1-86b1-7d43eecebf77.png)
![1](https://user-images.githubusercontent.com/106856709/171983313-9b4f329f-dc9d-4498-99c3-05f5cdd3c681.png)
![2](https://user-images.githubusercontent.com/106856709/171983603-673e1732-4ad8-45bf-b8f5-d03d50591e62.png)

**Comment by gmjosack** [comment]
Interesting. Is this observed in the png itself or after loading extracted assets back into the game? Would be good to isolate if this is from the DDS to png conversion or if it's happen on Playlunky when being converted back for loading into the game.

**Comment by iseeaitch** [comment]
Yes, and this seems to be appeared both in DDS to png conversion and Playlunky.
![3](https://user-images.githubusercontent.com/106856709/171986025-fa80916a-69d2-43bf-b9e7-d83bf84dc3a4.png)

**Comment by Dregu** [comment]
Thoughts from Discord:

1. DDS files in the game are premultiplied alpha, but extraction does not un-premultiply when making PNGs (premultiplied PNG is not even a thing but whatever)
2. Multiplying/Dividing the alpha is lossy and we can't really get an original PNG back that could be both loaded in PL and look right in an editor
3. Playlunky is premultiplying all images from mods, cause they are likely PNG and not premultiplied
4. Playlunky is also premultiplying the cached vanilla files it extracted when stitching entity sheets or sprite maps, which is bad and any unmodified parts get premultiplied twice in the end
5. Using any mods that custom sprite map to items.png will break all the other items because of 4
6. There might even be a feedback loop so files in the cache get premultiplied again and again

Solutions:

- Modlunky could predivide alphas when extracting, but it's not going to look the same when premultiplying again
  - We could just say the extracted PNG files are premultiplied and anyone who uses very transparent parts as base for their mods is stupid
- Playlunky definitely shouldn't divide or premultiply cached (vanilla) files ever, only premultiply user-provided files once before caching or stitching

I extracted the divided textures with this modification but nothing looks quite right either way...

```python
def dds_to_png(data):
    """Takes a .DDS `Image` and returns .png data."""
    img = Image.open(io.BytesIO(data))
    px = img.load()
    width, height = img.size
    for i in range(width):
        for j in range(height):
            if px[i, j][3] != 0:
                R = int(255.0 * px[i, j][0] / px[i, j][3] + 0.5)
                G = int(255.0 * px[i, j][1] / px[i, j][3] + 0.5)
                B = int(255.0 * px[i, j][2] / px[i, j][3] + 0.5)
                a = px[i, j][3]
                px[i, j] = (R, G, B, a)
    new_data = io.BytesIO()
    img.save(new_data, format="PNG")
    return new_data.getvalue()
```
